ttnn/cpp/ttnn/operations/data_movement/common/common.cpp

```diff
@@ -55,6 +55,84 @@ ttnn::Tensor pad_to_tile_vol(
     return tensor;
 }
 uint32_t wrap_index(int index, int size) { return index < 0 ? size + index : index; }
+
+ttnn::MemoryConfig create_sharded_memory_config(
+    const ttnn::Shape& shape,
+    const tt::tt_metal::CoreRangeSet& core_grid,
+    const ShardStrategy& strategy,
+    const tt::tt_metal::ShardOrientation& orientation,
+    bool halo,
+    bool use_height_and_width_as_shard_shape,
+    const tt::tt_metal::Layout& layout) {
+    auto is_tile_layout = layout == tt::tt_metal::Layout::TILE;
+
+    auto rank = shape.rank();
+    TT_FATAL(rank >= 2, "rank of tensor to shard must be at least 2.");
+
+    auto tensor_memory_layout = ttnn::TensorMemoryLayout::BLOCK_SHARDED;
+    if (strategy == ShardStrategy::WIDTH) {
+        tensor_memory_layout = ttnn::TensorMemoryLayout::WIDTH_SHARDED;
+    } else if (strategy == ShardStrategy::HEIGHT) {
+        tensor_memory_layout = ttnn::TensorMemoryLayout::HEIGHT_SHARDED;
+    }
+
+    auto shard_orientation = orientation;
+    auto shard_grid = core_grid;
+
+    auto height = shape[-2];
+    auto width = shape[-1];
+    std::array<uint32_t, 2> shard_shape;
+
+    if (use_height_and_width_as_shard_shape) {
+        if (shard_orientation == tt::tt_metal::ShardOrientation::ROW_MAJOR) {
+            shard_shape = {height, width};
+        } else if (shard_orientation == tt::tt_metal::ShardOrientation::COL_MAJOR) {
+            shard_shape = {width, height};
+        } else {
+            TT_THROW("Invalid shard orientation");
+        }
+    } else {
+        uint32_t batch_size = 1;
+        for (int i = 0; i < rank - 2; i++) {
+            batch_size *= shape[i];
+        }
+
+        auto tensor_height = batch_size * height;
+        auto tensor_width = width;
+        auto total_num_cores = shard_grid.num_cores();
+        auto grid_size = shard_grid.bounding_box().grid_size();
+
+        if (tensor_memory_layout == ttnn::TensorMemoryLayout::BLOCK_SHARDED) {
+            TT_ASSERT(grid_size.y * grid_size.x == total_num_cores, "Invalid CoreRangeSet for block sharding strategy");
+
+            if (shard_orientation == tt::tt_metal::ShardOrientation::ROW_MAJOR) {
+                auto tensor_height_padded =
+                    is_tile_layout ? tt::round_up(tensor_height, grid_size.y * 32) : tensor_height;
```

Comment:
Can you please replace 32s with an appropriate constant, like TILE_WIDTH?
Is it still okay to use 32 here in case we use tiny tiles?

Reply:
Ideally, all pieces of logic with rounding up to tiles should've been encapsulated within TensorLayout. @ayerofieiev-tt @TT-BrianLiu What do you think?

Reply:
Instead of 32 its definitely better to use tile size from tensor layout
I think same things can be queried from `tensor.get_tensor_spec.physical_size().height()`, returned value is basically your `tensor_height_padded` (I might mess up with func name here, but thats the idea)